pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java

```diff
@@ -643,6 +643,30 @@ public void libraryObjectInScript() throws Exception {
                 .go();
     }
 
+    @Issue("JENKINS-43035")
+    @Test
+    public void libraryObjectInWhenExpression() throws Exception {
+        otherRepo.init();
+        otherRepo.write("src/org/foo/Zot.groovy", "package org.foo;\n" +
```

Comment:
should probably test star and static imports as well in one script, to make sure they are added correctly.